src/Scout/ScoutEngine.php

```diff
@@ -0,0 +1,542 @@
+            $compound['mustNot'][] = ['in' => ['path' => $field, 'value' => $value]];
+        }
+
+        // Sort by field value only if specified
+        $sort = [];
+        foreach ($builder->orders as $order) {
+            $sort[$order['column']] = $order['direction'] === 'asc' ? 1 : -1;
+        }
+
+        $pipeline = [
+            [
+                '$search' => [
+                    'index' => self::INDEX_NAME,
+                    'compound' => $compound,
+                    'count' => ['type' => 'lowerBound'],
+                    ...($sort ? ['sort' => $sort] : []),
+                ],
+            ],
+            [
+                '$addFields' => [
+                    // Metadata field with the total count of documents
+                    '__count' => '$$SEARCH_META.count.lowerBound',
+                ],
+            ],
+        ];
+
+        if ($offset) {
+            $pipeline[] = ['$skip' => $offset];
+        }
+
+        if ($builder->limit) {
+            $pipeline[] = ['$limit' => $builder->limit];
+        }
+
+        $options = [
+            'typeMap' => ['root' => 'array', 'document' => 'array', 'array' => 'array'],
+        ];
+
+        return $collection->aggregate($pipeline, $options)->toArray();
+    }
+
+    /**
+     * Pluck and return the primary keys of the given results.
+     *
+     * @see Engine::mapIds()
+     *
+     * @param list<array|object> $results
+     */
+    #[Override]
+    public function mapIds($results): Collection
+    {
+        assert(is_array($results), new TypeError(sprintf('Argument #1 ($results) must be of type array, %s given', get_debug_type($results))));
+
+        return new Collection(array_column($results, '_id'));
+    }
+
+    /**
+     * Map the given results to instances of the given model.
+     *
+     * @see Engine::map()
+     *
+     * @param array $results
+     * @param Model $model
+     */
+    #[Override]
+    public function map(Builder $builder, $results, $model): Collection
+    {
+        assert(is_array($results), new TypeError(sprintf('Argument #2 ($results) must be of type array, %s given', get_debug_type($results))));
+        assert($model instanceof Model, new TypeError(sprintf('Argument #3 ($model) must be of type %s, %s given', Model::class, get_debug_type($model))));
+
+        if (! $results) {
+            return $model->newCollection();
+        }
+
+        $objectIds = array_column($results, '_id');
+        $objectIdPositions = array_flip($objectIds);
+
+        return $model->getScoutModelsByIds(
+            $builder,
+            $objectIds,
+        )->filter(function ($model) use ($objectIdPositions) {
+            return array_key_exists($model->getScoutKey(), $objectIdPositions);
+        })->map(function ($model) use ($results, $objectIdPositions) {
+            $result = $results[$objectIdPositions[$model->getScoutKey()]] ?? [];
+
+            foreach ($result as $key => $value) {
+                if ($key[0] === '_') {
```

Comment:
Wouldn't the `_id` field trigger this condition? If so, is that something you want set on the [scout metadata](https://github.com/laravel/scout/blob/v10.11.9/src/Searchable.php#L391)?

Noted that this was copied from other engines. AFAIK, the only metadata field used elsewhere is `__soft_deleted`. I'm curious as to why the engines don't check for a `__` prefix instead.

Reply:
I excluded `'_id'`, I don't want to make difference with other engine on this prefix.